File: src/main/java/fi/metatavu/metaform/server/persistence/dao/NumberReplyFieldDAO.java
package fi.metatavu.metaform.server.persistence.dao;

import java.util.UUID;

import javax.enterprise.context.ApplicationScoped;

import fi.metatavu.metaform.server.persistence.model.NumberReplyField;
import fi.metatavu.metaform.server.persistence.model.Reply;

/**
 * DAO class for NumberReplyField entity
 * 
 * @author Antti Leppä
 */
@ApplicationScoped
public class NumberReplyFieldDAO extends ReplyFieldDAO<NumberReplyField> {

  /**
   * Creates new NumberReplyField 
   * 
   * @param id id
   * @param reply reply
   * @param name field name
   * @param value field value
   * @return created field
   */
  public NumberReplyField create(UUID id, Reply reply, String name, Double value) {
    NumberReplyField replyField = new NumberReplyField(); 
    replyField.setId(id);
    replyField.setName(name);
    replyField.setReply(reply);
    replyField.setValue(value);
    return persist(replyField);
  }

  /**
   * Updates reply field 
   * 
   * @param replyField reply field
   * @param value value
   * @return updated field
   */
  public NumberReplyField updateValue(NumberReplyField replyField, Double value) {
    replyField.setValue(value);
    return persist(replyField);
  }
  
}

File: src/main/kotlin/fi/metatavu/metaform/persistence/dao/NumberReplyFieldDAO.kt
package fi.metatavu.metaform.persistence.dao

import fi.metatavu.metaform.persistence.model.NumberReplyField
import fi.metatavu.metaform.persistence.model.Reply
import java.util.*
import javax.enterprise.context.ApplicationScoped

/**
 * DAO class for NumberReplyField entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class NumberReplyFieldDAO : ReplyFieldDAO<NumberReplyField>() {

  /**
   * Creates new NumberReplyField
   *
   * @param id id
   * @param reply reply
   * @param name field name
   * @param value field value
   * @return created field
   */
  fun create(id: UUID, reply: Reply, name: String, value: Double): NumberReplyField {
    val replyField = NumberReplyField()
    replyField.id = id
    replyField.name = name
    replyField.reply = reply
    replyField.value = value
    return persist(replyField)
  }

  /**
   * Updates reply field
   *
   * @param replyField reply field
   * @param value value
   * @return updated field
   */
  fun updateValue(replyField: NumberReplyField, value: Double): NumberReplyField{
    replyField.value = value
    return persist(replyField)
  }
}